File: packages/firestore/ios/Plugin/Classes/Options/AddDocumentOptions.swift
import Foundation
import Capacitor
import FirebaseFirestore

@objc public class AddDocumentOptions: NSObject {
    private var reference: String
    private var data: [String: Any]

    init(reference: String, data: JSObject, firestore: Firestore) {
        self.reference = reference
        self.data = FirebaseFirestoreHelper.createHashMapFromJSObject(data, firestore: firestore)
    }

    func getReference() -> String {
        return reference
    }

    func getData() -> [String: Any] {
        return data
    }
}

File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/AddDocumentOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.firebase.firestore.FirebaseFirestoreHelper;
import java.util.Map;
import org.json.JSONException;

public class AddDocumentOptions {

    private String reference;
    private Map<String, Object> data;

    public AddDocumentOptions(String reference, JSObject data, @NonNull com.google.firebase.firestore.FirebaseFirestore firestore)
        throws JSONException {
        this.reference = reference;
        this.data = FirebaseFirestoreHelper.createHashMapFromJSONObject(data, firestore);
    }

    public String getReference() {
        return reference;
    }

    public Map<String, Object> getData() {
        return data;
    }
}
